### DISCUSSION TITLE
Tech Team Meeting - 2022-06-03

### DISCUSSION THREAD

**Discussion by jwildfire** [SafetyGraphics/safetyGraphics#677] [Meetings]
# Agenda

1. Framework Updates (@jwildfire  - 15 min)
   - safetyGraphics v2.1 - Metadata refactor
      - [Release notes](https://github.com/SafetyGraphics/safetyGraphics/releases/tag/v2.1.0)
      - [`?makeMeta`](https://safetygraphics.github.io/safetyGraphics/reference/makeMeta.html)
      - Example with custom Metadata - [1](https://github.com/SafetyGraphics/safetyGraphics/wiki/Cookbook#example-11---create-a-hello-world-data-domain-and-chart), [2](https://github.com/SafetyGraphics/safetyGraphics/wiki/ChartConfiguration#example-5---adding-a-chart-in-a-new-data-domain)
   - safetyCharts v0.3 - Metadata refactor + Widget Customization
      - [Release notes](https://github.com/SafetyGraphics/safetyCharts/releases/tag/v0.3.0) 
      - [`?safetyCharts::hepExplorer`](https://safetygraphics.github.io/safetyCharts/reference/hepExplorer.html)
   - safetyGraphics v2.2
      - Basic Chart-level requirements - [Pull Request](https://github.com/SafetyGraphics/safetyGraphics/pull/670)
      - [Other open issues](https://github.com/SafetyGraphics/safetyGraphics/milestone/24)
2. Best Practice Updates (15 min, Jeremy/Xiao)
   - [Filtering in safetyGraphics](https://github.com/SafetyGraphics/safetyGraphics/wiki/TechnicalFAQ#q-how-is-filteringsubsetting-handled-in-safetygraphics)
      - Participant-level properties -> DM (filter with filter tab) 
      - Domain-level data -> Domains (filter in individual charts)
   - [Baseline Calculations](https://github.com/SafetyGraphics/safetyGraphics/wiki/TechnicalFAQ#q-how-are-baseline-values-handled-in-safetygraphics)
   -  [*good practice* issues in {qtexplorer} ](https://github.com/SafetyGraphics/qtexplorer/discussions/44)
3. Workstream Updates (30 min)
   - Quick Updates (~2 min each)
      - [Volcano Plot](https://github.com/SafetyGraphics/volcanoPlot) (Jeremy) 
      - [Hep Explorer](https://github.com/SafetyGraphics/hep-explorer) (Jeremy)
      - [Nep Explorer](https://github.com/SafetyGraphics/nep-explorer) (Eli/Renan/Preston)
      - [BRATS](https://github.com/SafetyGraphics/BRForestPlot) (Brian)
   - Deep Dive - [QT](https://github.com/SafetyGraphics/qtexplorer) (Xiao/Rosie) - 15 min

**Comment by jwildfire** [comment]
@xni7 raised some good questions about the (somewhat clunky) UX for the current filter tab. I'm thinking there might be other approaches

@elimillera do you know how the filter sidebar is set up in https://github.com/RConsortium/submissions-pilot2? Any other options that people are aware of?

**Reply by xni7** [reply]
@jwildfire , some options to consider

- a modal window that can be activated on each chart
- bs4Dash type of "control bar" as seen in the [TTE example ](https://stefaneng.shinyapps.io/subpat-TTE). See screenshot below

![image](https://user-images.githubusercontent.com/31833867/171930684-09e09599-d532-4c50-aee3-be4b2c93b74d.png)

**Reply by samussiah** [reply]
I like this idea - perhaps throw the toggle by the population count using a [filter icon](https://fontawesome.com/icons/filter?s=solid).
